app/src/main/kotlin/com/wire/android/ui/home/conversations/model/messagetypes/audio/AudioMessageType.kt

```diff
@@ -0,0 +1,217 @@
+) {
+    Box(
+        modifier = Modifier
+            .padding(top = dimensions().spacing4x)
+            .background(
+                color = MaterialTheme.wireColorScheme.onPrimary,
+                shape = RoundedCornerShape(dimensions().messageAssetBorderRadius)
+            )
+            .border(
+                width = 1.dp,
+                color = MaterialTheme.wireColorScheme.secondaryButtonDisabledOutline,
+                shape = RoundedCornerShape(dimensions().messageAssetBorderRadius)
+            )
+            .padding(dimensions().spacing8x)
+    ) {
+        if (audioMediaPlayingState is AudioMediaPlayingState.Failed) {
+            FailedAudioMessage()
+        } else {
+            SuccessFullAudioMessage(
+                audioMediaPlayingState = audioMediaPlayingState,
+                totalTimeInMs = totalTimeInMs,
+                currentPositionInMs = currentPositionInMs,
+                onPlayButtonClick = onPlayButtonClick,
+                onSliderPositionChange = onSliderPositionChange,
+                onAudioMessageLongClick = onAudioMessageLongClick
+            )
+        }
+    }
+}
+
+@Composable
+private fun SuccessFullAudioMessage(
+    audioMediaPlayingState: AudioMediaPlayingState,
+    totalTimeInMs: Int,
+    currentPositionInMs: Int,
+    onPlayButtonClick: () -> Unit,
+    onSliderPositionChange: (Float) -> Unit,
+    onAudioMessageLongClick: (() -> Unit)? = null
+) {
+    val audioDuration by remember(currentPositionInMs) {
+        mutableStateOf(
+            AudioDuration(totalTimeInMs, currentPositionInMs)
+        )
+    }
+
+    val isFetching = audioMediaPlayingState is AudioMediaPlayingState.Fetching
+
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(dimensions().audioMessageHeight)
+            .clickable(Clickable(true, onLongClick = onAudioMessageLongClick)),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        WireSecondaryIconButton(
+            minWidth = 32.dp,
+            iconResource = getPlayOrPauseIcon(audioMediaPlayingState),
+            shape = CircleShape,
+            contentDescription = R.string.content_description_image_message,
+            state = if (isFetching) WireButtonState.Disabled else WireButtonState.Default,
+            onButtonClicked = onPlayButtonClick
+        )
+
+        Slider(
+            value = audioDuration.currentPositionInMs.toFloat(),
+            onValueChange = onSliderPositionChange,
+            valueRange = 0f..audioDuration.totalDurationInMs.toFloat(),
+            colors = SliderDefaults.colors(
+                inactiveTrackColor = colorsScheme().secondaryButtonDisabledOutline
+            ),
+            modifier = Modifier.weight(1f)
+        )
+
+        if (isFetching) {
+            WireCircularProgressIndicator(
+                progressColor = MaterialTheme.wireColorScheme.primary,
+                size = dimensions().spacing32x
+            )
```

Comment:
On a quick look at the designs, I'd say that these loading bar specs are not correct. I believe the size should be 16px and the color should be something like `Gray90` or maybe `secondaryButtonEnabled`. Better to address these now than in a design regression 😅